addons/web/static/src/notifications/notification.js

```diff
@@ -7,20 +7,8 @@ const { Component } = owl;
 export class Notification extends Component {
   setup() {
     this.notificationService = useService("notification");
-  }
-
-  get icon() {
-    switch (this.props.type) {
-      case "danger":
-        return "fa-exclamation";
-      case "warning":
-        return "fa-lightbulb-o";
-      case "success":
-        return "fa-check";
-      case "info":
-        return "fa-info";
-      default:
-        return this.props.icon;
+    if (this.props.onClose) {
+      owl.hooks.onWillUnmount(() => this.props.onClose());
```

Comment:
line 5, we typically do `const { Component, hooks } = owl;`, and thus here `hooks.onWillUnmount(...);`